typhon-core/src/lib.rs

```diff
@@ -63,6 +63,14 @@ const _: () = {
             CELL.set(settings)
                 .expect("Settings were already initalized")
         }
+        fn verify_password(&self, password: &[u8]) -> bool {
+            use argon2::{Argon2, PasswordHash, PasswordVerifier};
+            let password_hash =
+                PasswordHash::new(&self.password).expect("Unable to parse the password hash");
+            Argon2::default()
+                .verify_password(password, &password_hash)
```

Comment:
Parsing the argon2 string into a PasswordHash should happen before `Settings::init` imo, and the password's field of `Settings` should be a `PasswordHash`, not a `String`.